### PR TITLE
Add Delete Snapshot High Level REST API

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we add the delete snapshot API to the Java high level
REST client.

Relates #27205

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
Marked as "WIP" as we should wait for the create snapshot API so we can leverage it in tests (see https://github.com/elastic/elasticsearch/pull/31215).

**Comment by hub-cap** [comment]
@danielmitterdorfer you can use the low level rest API in the tests, and then change them once the high level API is done. This is what I did when implementing `GET /repositories` before `PUT`.

**Review by hub-cap — CHANGES_REQUESTED** [review]
This is a great PR, quite nice to review. Just fyi, running the doc build shows that there are failures to the docs. LGTM for the most part, once the test is impl'd and the docs build is fixed.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/SnapshotClient.java [hunk 1]
```diff
@@ -161,4 +163,34 @@ public void verifyRepositoryAsync(VerifyRepositoryRequest verifyRepositoryReques
...
+     * Deletes a snapshot.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html"> Snapshot and Restore
+     * API on elastic.co</a>
+     *
+     * @param deleteSnapshotRequest the request
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the response
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public DeleteSnapshotResponse deleteSnapshot(DeleteSnapshotRequest deleteSnapshotRequest, RequestOptions options) throws IOException {
```
Comment (javanna) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/SnapshotClient.java]: can you call these methods delete without the snapshot part? they will be called as client.snapshot().delete() then which is how they are defined in our SPEC

**Review by javanna — COMMENTED** [review]
left a small comment as well

File: client/rest-high-level/src/main/java/org/elasticsearch/client/SnapshotClient.java (hunk 1 above)
Comment (danielmitterdorfer) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/SnapshotClient.java]: Makes sense. I've changed it.

**Comment by danielmitterdorfer** [comment]
@hub-cap I've implemented the tests now based on the low-level client as per your suggestion and fixed the doc issues. Can you please have another look?

**Review by hub-cap — APPROVED** [review]
:ship:

**Comment by hub-cap** [comment]
Doh that was supposed to be a :shipit: not a :ship:

**Comment by danielmitterdorfer** [comment]
Both icons work for me. ;) Thanks for the review!

**Comment by danielmitterdorfer** [comment]
Backported to 6.x in 71b3ac2.

**Pull request merged**